src/SFA.DAS.Forecasting.Commitments.Functions/NServicebusTriggerFunctions/ApprenticeshipCompletedFunction.cs

```diff
@@ -9,23 +14,54 @@ namespace SFA.DAS.Forecasting.Commitments.Functions.NServicebusTriggerFunctions
     public class ApprenticeshipCompletedFunction
     {
         private readonly IForecastingDbContext _forecastingDbContext;
+        private readonly IMapper _mapper;
+        private readonly ICommitmentsApiClient _commitmentsApiClient;
+        private readonly ILogger _logger;
 
-        public ApprenticeshipCompletedFunction(IForecastingDbContext forecastingDbContext)
+        public ApprenticeshipCompletedFunction(IForecastingDbContext forecastingDbContext, 
+            ICommitmentsApiClient commitmentsApiClient,
+            IMapper mapper,
+            ILogger<ApprenticeshipCompletedFunction> logger)
         {
             _forecastingDbContext = forecastingDbContext;
+            _commitmentsApiClient = commitmentsApiClient;
+            _mapper = mapper;
+            _logger = logger;
         }
 
         [FunctionName("ApprenticeshipCompleted")]
         public async Task Run(
             [NServiceBusTrigger(Endpoint = "SFA.DAS.Fcast.ApprenticeshipCompletedEvent")] ApprenticeshipCompletedEvent message)
         {
-            var selectedApprenticeship = _forecastingDbContext.Commitment.FirstOrDefault(x => x.ApprenticeshipId == message.ApprenticeshipId);
-            if (selectedApprenticeship != null)
+            try
             {
-                selectedApprenticeship.ActualEndDate = message.CompletionDate;
-                selectedApprenticeship.Status = Status.Completed;
+                _logger.LogInformation($"Apprenticeship Completed function executing at: [{DateTime.UtcNow}] UTC, event with ApprenticeshipId: [{message.ApprenticeshipId}].");               
```

Comment:
Can we move the logic into a Handler like the other function app. This would simplify tests as you wouldn't be required to pass the Azure Functions specific params like TimerInfo etc

Reply:
done